libmogwai-schedule/scheduler.c

```diff
@@ -147,7 +151,7 @@ static void
 mws_scheduler_class_init (MwsSchedulerClass *klass)
 {
   GObjectClass *object_class = (GObjectClass *) klass;
-  GParamSpec *props[PROP_PEER_MANAGER + 1] = { NULL, };
+  GParamSpec *props[PROP_ALLOW_DOWNLOADS + 1] = { NULL, };
```

Comment:
You could keep a const variable `last_scheduler_property` pointing to the last prop (or `num_scheduler_properties` with the sum already), and update it in the future, which would be self-documenting. Either way, it'd just get a bit more elegant IMO, but not critical to understanding what's going on.

Reply:
I think this works just about well enough; the compiler should warn if you don’t update the array length but do assign to the new (`PROP_ALLOW_DOWNLOADS`) element in the array.